contrib/terraform/openstack/modules/network/main.tf

```diff
@@ -0,0 +1,24 @@
+
+resource "openstack_networking_router_v2" "k8s" {
+  name             = "${var.cluster_name}-router"
+  admin_state_up   = "true"
+  external_gateway = "${var.external_net}"
+}
+
+resource "openstack_networking_network_v2" "k8s" {
+  name           = "${var.network_name}"
+  admin_state_up = "true"
+}
+
+resource "openstack_networking_subnet_v2" "k8s" {
+  name            = "${var.cluster_name}-internal-network"
+  network_id      = "${openstack_networking_network_v2.k8s.id}"
+  cidr            = "10.0.0.0/24"
+  ip_version      = 4
+  dns_nameservers = ["8.8.8.8", "8.8.4.4"]
```

Comment:
it's internal network, should dns be in vars to be configurable? what if internal names is needed? (not an openstack expert, but this seems wrong)